ebmlite/util.py

```diff
@@ -31,6 +31,56 @@
 # ==============================================================================
 
 
+def createID(idClass, schema, exclude=(), min=0x80, max=0x1FFFFFFF, count=1):
+    """ Generate unique EBML IDs. Primarily intended for use 'offline' by
+        humans creating EBML schemata.
+
+        :param idClass: The EBML class of ID, one of (case-insensitive):
+            * `'a'`: Class A (1 octet, base 0x8X)
+            * `'b'`: Class B (2 octets, base 0x4000)
+            * `'c'`: Class C (3 octets, base 0x200000)
+            * `'d'`: Class D (4 octets, base 0x10000000)
+        :param schema: The `Schema` in which the new IDs must coexist.
+        :param exclude: A list of additional IDs to avoid.
+        :param min: The minimum ID value, within the ID class' range.
+        :param max: The maximum ID value, within the ID class' range.
+        :param count: The maximum number of IDs to generate. The result may be
+            fewer than specified if too few meet the given criteria.
+        :returns: A list of EBML IDs that match the given criteria.
+    """
+    ranges = dict(a = (0x80, 0xFF),
+                   b = (0x4000, 0x7FFF),
+                   c = (0x200000, 0x3FFFFF),
+                   d = (0x10000000, 0x1FFFFFFF))
+    idc = idClass.lower()
+    if idc not in ranges:
+        raise KeyError('Invalid ID class %r: must be one of %r' %
+                       (idClass, list(ranges)))
+
+    # Keep range within the one specified and the one imposed by the ID class
+    idrange = ranges[idc]
+    idrange = (__builtins__['max'](idrange[0], min),
+               __builtins__['min'](idrange[1], max))
+
+    exclude = list(exclude)
+    exclude.extend(schema.elements.keys())
+
+    result = []
+    for i in range(*idrange):
+        if i in exclude:
+            continue
+        if len(result) == count:
+            break
+        result.append(i)
+
+    return result
```

Comment:
this can be a one-liner (excluding the import):

```python
import itertools

return itertools.islice((i for i in range(*idrange) if i not in exclude), count)
```

Reply:
I'm not sure that the gain is worth importing another library, but I did half the length of this bit in commit ebf540431cee302d6023d4dabfcb6f5906d64c4f.

Reply:
You don't have to get into it, but ftr I don't understand the downside of using built-in packages; they just make life easier.

Reply:
I want to keep `ebmlite` light. Even standard packages need to be imported, taking time and consuming memory. Not very much of either, admittedly, but it still outweighs the return in this case. The code is also more easily readable without it.

Reply:
I... don't agree with anything you just said.

> I want to keep ebmlite light. Even standard packages need to be imported

If imports are a concern, then I assume unused imports are equally as concerning, like the one I noted on this PR in https://github.com/MideTechnology/ebmlite/pull/56#discussion_r597798573. I made a new PR (#58) to add flake8 to our Travis tests for this repo, which I've configured to check for purely logical errors such as unused imports; please review.

> taking time and consuming memory. Not very much of either, admittedly

that's an understatement. the fact that we're splitting hairs on **micro**seconds of time (try running `import timeit; timeit.timeit("import itertools", number=1)`) makes this conversation especially frustrating.

> The code is also more easily readable without it.

I just disagree.